### PR TITLE
Remove aliases from pkg/config and refactor all references to types/functions

### PR DESCRIPTION
**Pull request by jackgopack4**
### What does this PR do?

replaces all existing references to `pkg/config` with the respective sub-module. The majority of functions and variables were already moved to aliases with #20034; this change removes obfuscation and makes it clearer what our code is doing.

Additionally, remaining functions that were in pkg/config were moved to appropriate submodules, such as pkg/config/legacy and pkg/config/setup

### Motivation

closes https://datadoghq.atlassian.net/browse/OTEL-1134

### Additional Notes

legacy_converter.go was moved from pkg/config to pkg/config/legacy

Ideally I would have broken this up into smaller PRs, but refactoring the pkg/config/legacy code was causing a circular dependency on every package importing pkg/config so I went through and updated all of them instead.

### Possible Drawbacks / Trade-offs

slight formatting/style preferences used across the codebase; could change import names to match local code style. Additionally if a function/functions belong in a different pkg/config submodule I would be happy to move it where makes sense.

### Describe how to test/QA your changes

run unit/integration tests locally and through CI

### Reviewer's Checklist

- [X] If known, an appropriate milestone has been selected; otherwise the Triage milestone is set.
- [ ]  Use the major_change label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [X]  A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the changelog/no-changelog label has been applied.
- [X]  Changed code has automated tests for its functionality.
- [ ]  Adequate QA/testing plan information is provided if the qa/skip-qa label is not applied.
- [X]  At least one team/.. label has been applied, indicating the team(s) that should QA this change.
- [ ]  If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ]  If applicable, the need-change/operator and need-change/helm labels have been applied.
- [ ]  If applicable, the k8s/<min-version> label, indicating the lowest Kubernetes version compatible with this feature.
- [ ]  If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/legacy/converter.go
Comment (jackgopack4) [inline pkg/config/legacy/converter.go:1]: Changes to name made to pass linter ("stuttering" name legacy.LegacyConfigConverter => legacy.ConfigConverter)

File: pkg/config/setup/constants/constants.go [hunk 1]
```diff
@@ -9,4 +9,6 @@ package constants
 const (
 	// DefaultEBPFLessProbeAddr defines the default ebpfless probe address
 	DefaultEBPFLessProbeAddr = "localhost:5678"
+	// ClusterIDCacheKey is the key name for the orchestrator cluster id in the agent in-mem cache
+	ClusterIDCacheKey = "orchestratorClusterID"
```
Comment (jackgopack4) [inline pkg/config/setup/constants/constants.go:13]: relocated from pkg/config/consts.go

File: pkg/config/test_helpers.go
Comment (jackgopack4) [inline pkg/config/test_helpers.go:1]: aliases only; no longer needed when imports of pkg/config are removed

File: pkg/flare/archive.go [hunk 2]
```diff
@@ -45,6 +45,11 @@ var cnfFileExtRx = regexp.MustCompile(`(?i)\.ya?ml`)
 // searchPaths is a list of path where to look for checks configurations
 type searchPaths map[string]string
 
+// GetProcessAPIAddressPort Alias using Datadog config
```
Comment (jackgopack4) [inline pkg/flare/archive.go:48]: required in order to pass the function to getChecksFromProcessAgent on L71

File: pkg/config/legacy_converter.go
Comment (jackgopack4) [inline pkg/config/legacy_converter.go:1]: functions moved into pkg/config/legacy

**Comment by jackgopack4** [comment]
not sure why the Serverless Binary Size test is failing; seems to be due to an external comment integration. Serverless binary size decreases by 128 bytes with this PR

**Review by yshapiro-57 — APPROVED** [review]
This is fine from the telemetry and analytics perspective, changes to collector.go and collector_test.go look reasonable.

**Review by paulcacheux — APPROVED** [review]
LGTM for agent-security owned files

**Review by steveny91 — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: pkg/serverless/trace/trace.go [hunk 3]
```diff
@@ -11,19 +11,20 @@ import (
 	"os"
 	"strings"
 
+	"github.com/DataDog/datadog-go/v5/statsd"
+
```
Comment (purple4reina) [inline pkg/serverless/trace/trace.go:15]: nit, this probably didn't need to move

**Review by purple4reina — APPROVED** [review]

File: pkg/serverless/trace/trace.go (hunk 3 above)
Comment (jackgopack4) [reply pkg/serverless/trace/trace.go:15]: agreed, but I have the linters enabled for pre-commit hooks so it was "easier" to let this change happen. Happy to change it back if you prefer.

**Review by derekwbrown — APPROVED** [review]
Approved for (3) (out of 584?) wkit-owned files

File: pkg/serverless/trace/trace.go (hunk 3 above)
Comment (purple4reina) [reply pkg/serverless/trace/trace.go:15]: No worries, thanks for the explanation!

**Review by jmw51798 — APPROVED** [review]

**Review by lrbhatia — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Review by BaptisteFoy — APPROVED** [review]
LGTM for RC/FA owned files

**Review by chouetz — APPROVED** [review]
ok for devx-infra owned files

**Review by vickenty — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]

**Review by GustavoCaso — APPROVED** [review]
Amazing work 🎉 

Love that we remove the alias inderection 👏 

As a note I noticed that in some functions we reference  the function call `pkgconfigsetup.Datadog()` multiple times. Is not a big deal, but would be nice if we could hoist that into a variable. Especially, since that function call deals with a read lock, so any unnecessary use of the lock 🔒 is not a good approach

I'm not sure if this suggestion would apply for all cases, but would be worth checking at least 😃 

```go
// Datadog returns the current agent configuration
func Datadog() pkgconfigmodel.Config {
	datadogMutex.RLock()
	defer datadogMutex.RUnlock()
	return datadog
}
```

**Comment by jackgopack4** [comment]
> As a note I noticed that in some functions we reference the function call `pkgconfigsetup.Datadog()` multiple times. Is not a big deal, but would be nice if we could hoist that into a variable. Especially, since that function call deals with a read lock, so any unnecessary use of the lock 🔒 is not a good approach

I like this idea; I'll look into this and consider opening a second PR; want to keep this PR as close to a 1:1 match of existing code as possible

**Review by jszwedko — APPROVED** [review]
Changes to @DataDog/agent-processing-and-routing files look straightforward.

**Review by kangyili — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]
LGTM for @DataDog/platform-integrations.

nit: could be good to refactor the PR title 😅

**Review by gjulianm — APPROVED** [review]

**Review by lu-zhengda — APPROVED** [review]

**Review by davidor — APPROVED** [review]
:+1: for container-platform files

**Review by daniel-taf — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Review by jonbodner — APPROVED** [review]
I'm approving, but this should have been a single module, not multiple modules. Also, while the changes work, one of the reasons why Go allows you to specify an override name for an import is so you can keep the previous name when you change the import. If that had been done, the only changes in the files would have been the import line. It would have meant more custom changes in each file, but it would have been fewer changes to review.

**Pull request merged**